lib/sax-machine/handlers/sax_abstract_handler.rb

```diff
@@ -65,6 +65,7 @@ def _start_element(name, attrs = [])
             case element_config.data_class.to_s
             when "Integer" then 0
             when "Float"   then 0.0
+            when "BigDecimal" then BigDecimal.new("0.0")
```

Comment:
I know this change is from a while ago, but the API changed to `BigDecimal("0.0")` [Ruby change](https://github.com/ruby/bigdecimal/commit/533737338db915b00dc7168c3602e4b462b23503)